ZuluContent/Zulu/Engines/Magic/Enchantments/Buffs/MagicReflection.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using MessagePack;
+using Server;
+using Server.Items;
+using Server.Spells;
+using ZuluContent.Zulu.Engines.Magic.Enums;
+
+namespace ZuluContent.Zulu.Engines.Magic.Enchantments.Buffs
+{
+    [MessagePackObject]
+    public class MagicReflection : Enchantment<MagicReflectionInfo>, IBuff
+    {
+        private string m_Description;
+        [IgnoreMember] public override string AffixName => string.Empty;
+
+        [Key(1)] public int Value { get; set; } = 1;
+
+        [Key(2)] public SpellCircle Circle { get; set; } = SpellCircle.First;
+
+        #region IBuff
+
+        [IgnoreMember] public BuffIcon Icon { get; init; } = BuffIcon.MagicReflection;
+        [IgnoreMember] public string Title { get; init; } = "Magic Reflection";
+
+        [IgnoreMember]
+        public string Description
+        {
+            get => m_Description ??= "Reflects the next harmful spell";
+            init => m_Description = value;
+        }
+
+        [IgnoreMember] public string[] Details { get; init; }
+        [IgnoreMember] public bool ExpireOnDeath { get; init; } = true;
+        [IgnoreMember] public bool Dispellable { get; init; } = false;
+        [IgnoreMember] public TimeSpan Duration { get; init; }
+        [IgnoreMember] public DateTime Start { get; init; }
+        
+        // We should use this first before checking item enchantments
+        [CallPriority(1)]
+        public override void OnCheckMagicReflection(Mobile parent, Spell spell, ref bool reflected)
+        {
+            if (parent != spell?.Caster && spell is ITargetableAsyncSpell<Mobile> && !reflected)
```

Comment:
Wondering if we need to make changes to the PermanentMagicReflect and MagicReflect (with charges) enchantments to be aligned with this?

Reply:
Yeah they would work better using this system, possibly by extending the `MagicReflection` buff itself but I might leave that fo a future PR separate from the spells. I need to stop getting distracted or I'll end up refactoring everything!